### PR TITLE
Editor: Extract BlockItem as a reusable component

### PR DESCRIPTION
**Pull request by gziolo**
## Description

This PR extracts `BlockListBlock` component and renames to `BlockItem` to reflect the fact that it's exposed to hooks as `editor.BlockItem`.

## How Has This Been Tested?
Manually, unit tests, e2e tests.

## Types of changes
Refactoring. No visual changes.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: blocks/editable/index.js [hunk 1]
```diff
@@ -436,7 +436,7 @@ export default class Editable extends Component {
 		// Find the parent "relative" positioned container
 		const container = this.props.inlineToolbar ?
 			this.editor.getBody().closest( '.blocks-editable' ) :
-			this.editor.getBody().closest( '.editor-block-list__block' );
+			this.editor.getBody().closest( '.editor-block-item' );
```
Comment (youknowriad) [inline blocks/editable/index.js]: We should definitely try to find an alternative here, we're referencing a classname created in another module.

**Comment by youknowriad** [comment]
Trying to understand the reasoning behind this change, because I don't expect this component to be used outside the editor module.

Also, should we move the other components used by this one into the same folder, things like `block-contextual-toolbar`, `block-html`, `block-crash-boundary`... These components are defined in the `block-list` folder but they're used in `block-item`

**Comment by gziolo** [comment]
> Trying to understand the reasoning behind this change, because I don't expect this component to be used outside the editor module.

@mtias mentioned in another PR that we could pick a different name for `BlockListBlock` and suggested `BlockItem`. @youknowriad you share valid concerns. I didn't think too much about that when I decided to do this simple refactoring. The thing is that to make it happen I had to move those files to their own folder. If you don't like the fact that we are exposing a component that is not meant to be consumed outside of the editor, let me propose renaming the file to `item.js` and make the component name `BlockListItem` instead. Would that work for you? I'm not that super attached to this change, so I can also close it if you have any concerns about that. We just need to make sure that we expose hooks for this component under the name that aligns with the component's name itself.

**Comment by youknowriad** [comment]
>  let me propose renaming the file to item.js and make the component name BlockListItem instead. Would that work for you

That sounds good to me. I'm not against having it in a separate folder, but this might be a good first step to avoid moving all the underlying components to this new folder.

**Comment by gziolo** [comment]
Thanks @youknowriad for your input. I will open a new PR with the file name change only as discussed above 👍

**Pull request closed without merging**